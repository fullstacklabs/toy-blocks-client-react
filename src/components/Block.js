import React from "react";
import PropTypes from "prop-types";
import { Box, makeStyles, Typography } from "@material-ui/core";
import colors from "../constants/colors";

const Block = ({ block }) => {
  const classes = useStyles();
  return (
    <Box className={classes.box}>
      <div className={classes.block}>
        <Typography className={classes.index}>
          {String(block.attributes.index).padStart(3, '0')}
        </Typography>
        <Typography className={classes.data}>
          {block.attributes.data}
        </Typography>
      </div>
    </Box>
  );
};

const useStyles = makeStyles(() => ({
  box: {
    paddingBottom: '4px'
  },
  block: {
    background: 'rgba(0, 0, 0, 0.12)',
    paddingBottom: '4px',
    padding: '8px',
    font: 'Roboto'
  },
  index: {
    color: colors.blue,
    fontSize: '10px',
    fontWeight: 700,
    letterSpacing: '2px'
  },
  data: {
    fontSize: '14px',
    paddingBottom: '4px'
  }
}));

Block.propTypes = {
  block: PropTypes.shape({
    id: PropTypes.string,
    type: PropTypes.string,
    attributes: PropTypes.shape({
      index: PropTypes.number,
      timestamp: PropTypes.number,
      data: PropTypes.string,
      "previous-hash": PropTypes.string,
      hash: PropTypes.string
    }),
  }).isRequired,
};

export default Block;
