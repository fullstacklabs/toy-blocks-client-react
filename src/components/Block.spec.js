import { create } from "react-test-renderer";
import React from "react";
import Block from "./Block";

describe("<Block />", () => {
  it("should match snapshot", () => {
    const component = create(<Block attributes={{ index: 1, data: "Some descript" }} />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
