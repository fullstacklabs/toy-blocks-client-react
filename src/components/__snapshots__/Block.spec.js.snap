// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Block /> should match snapshot 1`] = `
<div
  className="MuiBox-root MuiBox-root-5 makeStyles-panel-3"
>
  <sub
    className="makeStyles-sub-1"
  >
    001
  </sub>
  <p
    className="makeStyles-detail-2"
  >
    Some descript
  </p>
</div>
`;
